shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/providers/CustomerBridgeNamespaceProviderImpl.java

```diff
@@ -62,8 +63,11 @@ public void deleteNamespaceIfEmpty(Namespace namespace) {
             // deletion can be tricky with finalizers, pending objects, etc.
             // let's start simple and build on top of the constraints and scenarios as we evolve.
             if (!kubernetesClient.namespaces().withName(namespace.getMetadata().getName()).delete()) {
-                LOGGER.warn("Namespace '{}' hasn't been deleted", namespace.getMetadata().getName());
+                LOGGER.warn("Namespace '{}' has not been deleted.", namespace.getMetadata().getName());
             }
+        } else {
+            LOGGER.warn("Namespace '{}' has not been deleted as it contained other resources.", namespace.getMetadata().getName());
```

Comment:
I _should_ be able to delete the [scheduled](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1070/files#diff-59cb844d16bf6e217094b6f0b48152d3ca0cf3fc2c5e258e05ab8f3f011afad9R48) method and the `deleteNamespaceIfEmpty(..)` [method](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1070/files#diff-59cb844d16bf6e217094b6f0b48152d3ca0cf3fc2c5e258e05ab8f3f011afad9R61). 

Would you agree?

They both serve as "house keeping" on `main` to delete a `Namespace` when all of it's `BridgeIngress` Custom Resources are removed (that should not be required now we have a `Namespace` per `Bridge`).

Reply:
@manstis We need to consider `namespaces` with existing bridges in them. Ideally we'd clean them up over time as well. 

Just walking through the code, one proposal that came to mind:

1. Rename this method `deleteNamespace`
2. Call it from the `BridgeIngressController.cleanup` method when a `BridgeIngress` is deleted, taking the `namespace` to delete from the `BridgeIngress` resource
3. Replace the logic in this method to ensure that the `namespace` is marked for deletion _if_:
   1. There are no `BridgeIngress` in the `namespace`, or 
   2. There are `BridgeIngress` in the `namespace` but they all have the `HasMetaData.isMarkedForDeletion` set to true
4. Delete the method annotated with `@Scheduled` in this class.

That should mean that over time we eventually clear-up the namespaces as we delete Bridges using the old format and it should work equally as well for `namespaces` created in the new format.